### DISCUSSION TITLE
Somehow I cannot import  { vite as vidstack } from '@vitejs/plugin-vue'

### DISCUSSION THREAD

**Discussion by hnuecke** [vidstack/player#1175] [Q&A]
I'm using IntelliJ Idea and setup a new project with vue3 and TS.
I followed the player doc and everthing went smoothly up to the point "9 Setup Bundler" where I wanted to make the propsed import:
"import { vite as vidstack } from '@vitejs/plugin-vue';"

I get an error message telling me that '@vitejs/plugin-vue' has no exported member 'vite'.

What did I miss? I'm pretty new with vue, ts and npm. SO hopefully this is not a completely stupid question ;-)

Thanks for any hint to solve this issue!

**Comment by mihar-22** [comment]
👋 

It should be `import { vite as vidstack } from "vidstack/plugins"`, so replace `"@vitejs/plugin-vue"` with `"vidstack/plugins"`. I double checked the docs incase there was a mistake but I didn't see it. Let me know if we made a mistake somewhere and I'll fix it asap :)

**Reply by hnuecke** [reply]
Sorry, I should have mentioned that I tried that first. But I got the message:
![2024-02-20_13-19-48](https://github.com/vidstack/player/assets/3041318/924ae999-251b-4f0f-928d-7f32d68b0ccf)
Vue: Cannot find module vitejs/plugin-vue or its corresponding type declarations.

I imported with npm, if that's important to know.

**Reply by mihar-22** [reply]
Have you installed the package? Run the following in your terminal: “npm install @vitejs/plugin-vue”

**Reply by hnuecke** [reply]
I checke my nodes-modules repository of my project and can find a package "vidstack". But there indeed is nothing like "plugin-vue" in it.
And btw I also have a package named @vite with a subfolder plugin-vue. But in ./dist there are only index files.

Here the package.json of  @vitejs/plugin-vue::
[package.json](https://github.com/vidstack/player/files/14344920/package.json)

I could not finish a "npm i vidstack" because I had to enter mit githaub credientials, but the localhost address that opened in the browser was (of course) refused and I got stucked.
I downloaded the zip package but also in that package I couldn't find a plugin-in package/file?!

If it helps here the package json I have at the node-modules level:
[package.json](https://github.com/vidstack/player/files/14347675/package.json)

Perhaps I messed things up and should start from scratch; and with vidstack first?!
I feel like I am somehow lost ;-)

Thanks

**Reply by hnuecke** [reply]
I tried again installing the package as proposed, but to no avail:
![vidstack problem](https://github.com/vidstack/player/assets/3041318/0b4a424d-d9c8-4e21-b4e7-1138cc1a2d2a)

**Reply by hnuecke** [reply]
I 'm not sure if you can see this stackblitz view (it's the first time I'm using it ;-):
https://stackblitz.com/~/github.com/hnuecke/vue-mz3

But if so (I made my repo public for the time being) I think this could be the fastest way to find out what's wrong in my setup!

**Reply by hnuecke** [reply]
You immediately should see the problem, as an error message should pop up.
And you can check the settings in vite.config.ts

**Reply by hnuecke** [reply]
I built another project, with all default settings and not as a typescript project.
This time things look better ;-)
I can imagine that I did not fully understand everything needed for a typescript environment, e.g. I added settings to vite.config.ts  instead of vite.config.js, what might have been an incorrect approach.
I will try proceeding with this minimalistic project and learning by doing vue3 and vidstack first, before I attempt to also get into typscript.
Typescript is attractive because of Types that I know from my ActionScript3 developments. But it's not a good idea learning everything at the same time ;-)

If you immediately can identify my problem(s) now with all the access I provided, I would appreciate.
But I'm also OK with putting that aside and better coming back when I run into the next problem.

Thank you so much for your immediate support; excellent experience and encouraging.

**Reply by mihar-22** [reply]
I can't open the StackBlitz as it's private I believe but make sure you've installed `vidstack@next`, that's probably why it's not working.

**Reply by hnuecke** [reply]
OK, I'll try and let you know!

**Reply by hnuecke** [reply]
npm didn't find the file, although vidstack is installed. I guess it's a dead end road and I first should learn a bit more about the basics without taking your time for such beginner's questions ;-)
Here's the latest error log:
 C:\Users\hans\AppData\Local\npm-cache\_logs\2024-02-21T08_56_46_950Z-debug-0.log
 
 I really think I messed things up when installing all those packages (that I for now not even do need).
 Probably some packages had been installed as DevDependency by mistake or...?
 
 Never mind, I think we can close this, since the new project I set up (without typescript and all the stuff not needed at the moment) is working ;-)

Many thanks for your patience and help!
